### DISCUSSION TITLE
Heat exchanger behavior?

### DISCUSSION THREAD

**Discussion by jdybvik** [BeamCtrl/Airiana#34] [General]
Hi. As the outdoor temperatur is sinking, I've been wondering about the heat exchanger behavior. In this scenario; with inlet temperatur < 5c, and extract temperature <  target - 1 c  - I figure the heat exchanger should be forced on permanently. However, the heat exchanger cycles on and off fairly often, leaving the temperature efficiency around 82% (and never supply temperatures above 18,2c). 

What might be causing this behavior? Is it an error? AI-sensor?

<img width="709" alt="Skjermbilde 2023-10-09 kl  10 24 40" src="https://github.com/BeamCtrl/Airiana/assets/11717743/b61b4615-659a-4bd8-ad79-cc8d126572b7">

**Comment by BeamCtrl** [comment]
Yes, it should be on with no interruptions at those temperatures. Have you checked the log file in RAM/err on why the cycling occurs?
//Daniel

**Reply by BeamCtrl** [reply]
seems theres a bug in the code that can cause this behaviour for non Savecair units, pushed a new commit to mitigate.[28662b0](https://github.com/BeamCtrl/Airiana/commit/28662b011904d355927cb0a791802d860bfe366c)

**Reply by jdybvik** [reply]
Thanks for looking in to it. Nothing out of the ordinary in err, as far as I can gather: 

('127.0.0.1', 47985):9 at       Tue Oct 10 15:38:05 2023
Fire start at:  Tue Oct 10 15:38:05 2023
Pressure difference set to: -10%        Tue Oct 10 15:38:05 2023
Changing fanspeed to:3          Tue Oct 10 15:38:06 2023
Changing fanspeed to:1          Tue Oct 10 15:48:07 2023
Pressure difference set to: 10% Tue Oct 10 15:48:09 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Changing fanspeed to:3          Tue Oct 10 19:31:53 2023
Engaged Shower mode at  Tue Oct 10 19:31:54 2023
Leaving Shower mode dT: 0.2990716353699412      Tue Oct 10 19:38:43 2023
Changing fanspeed to:1          Tue Oct 10 19:38:43 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Changing fanspeed to:3          Wed Oct 11 05:47:08 2023
Engaged Shower mode at  Wed Oct 11 05:47:09 2023
Leaving Shower mode dT: 0.3005474384087172      Wed Oct 11 05:51:01 2023
Changing fanspeed to:1          Wed Oct 11 05:51:01 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
('127.0.0.1', 36919):9 at       Wed Oct 11 14:57:48 2023
Fire start at:  Wed Oct 11 14:57:48 2023
Pressure difference set to: -10%        Wed Oct 11 14:57:48 2023
Changing fanspeed to:3          Wed Oct 11 14:57:49 2023
Changing fanspeed to:1          Wed Oct 11 15:07:50 2023
Pressure difference set to: 10% Wed Oct 11 15:07:52 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.

Have update Airiana now, but unfortunately the exchanger still seem to cycle on/off as before. 

Jo

**Comment by BeamCtrl** [comment]
There is definetly an issue with the exchanger handling on non-savecair units still there, I will look into it tomorrow
/Daniel

14 okt. 2023 kl. 18:49 skrev jdybvik ***@***.***>:

﻿

Thanks for looking in to it. Nothing out of the ordinary in err, as far as I can gather:

('127.0.0.1', 47985):9 at Tue Oct 10 15:38:05 2023
Fire start at: Tue Oct 10 15:38:05 2023
Pressure difference set to: -10% Tue Oct 10 15:38:05 2023
Changing fanspeed to:3 Tue Oct 10 15:38:06 2023
Changing fanspeed to:1 Tue Oct 10 15:48:07 2023
Pressure difference set to: 10% Tue Oct 10 15:48:09 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Changing fanspeed to:3 Tue Oct 10 19:31:53 2023
Engaged Shower mode at Tue Oct 10 19:31:54 2023
Leaving Shower mode dT: 0.2990716353699412 Tue Oct 10 19:38:43 2023
Changing fanspeed to:1 Tue Oct 10 19:38:43 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Changing fanspeed to:3 Wed Oct 11 05:47:08 2023
Engaged Shower mode at Wed Oct 11 05:47:09 2023
Leaving Shower mode dT: 0.3005474384087172 Wed Oct 11 05:51:01 2023
Changing fanspeed to:1 Wed Oct 11 05:51:01 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
('127.0.0.1', 36919):9 at Wed Oct 11 14:57:48 2023
Fire start at: Wed Oct 11 14:57:48 2023
Pressure difference set to: -10% Wed Oct 11 14:57:48 2023
Changing fanspeed to:3 Wed Oct 11 14:57:49 2023
Changing fanspeed to:1 Wed Oct 11 15:07:50 2023
Pressure difference set to: 10% Wed Oct 11 15:07:52 2023
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.
Holdoff time in effect, will re-ping in one hour.

Have update Airiana now, but unfortunately the exchanger still seem to cycle on/off as before.

Jo

**Comment by BeamCtrl** [comment]
new fix is pushed along with unit testing framework for testing stuff. The unit test found a bug and its now fixed, I sure hope ;)
